hyperspy/roi.py

```diff
@@ -486,6 +490,28 @@ def __call__(self, signal, out=None, axes=None):
         return s
 
 
+def guess_vertical_or_horizontal(axes, signal):
+    # Figure out whether to use horizontal or veritcal line:
```

Comment:
To be more consistent with RangeWidget, I would suggest that this function determine the "movement direction" of the widget rather than the line direction.

Reply:
I don't get it, shouldn't it be the same?

Reply:
The following code should illustrate what I mean.
```python
import hyperspy.api as hs
import numpy as np

s = hs.signals.Signal2D(np.random.random((100, 1000)))
s.plot()

roi = hs.roi.SpanROI(10, 50)
w = roi.add_widget(s, axes=[s.axes_manager.signal_axes[1]])
# `guess_vertical_or_horizontal` returns `horizontal`
# RangeWidget is instantiated with direction=`vertical`
assert w.span.direction == 'vertical'

roi2 = hs.roi.Point1DROI(5)
w2 = roi2.add_widget(s, axes=[s.axes_manager.signal_axes[1]])
# `guess_vertical_or_horizontal` returns `horizontal`
# HorizontalLineWidget is instantiated
assert isinstance(w2, hs.hyperspy.drawing.widgets.HorizontalLineWidget)
```
The current definition of direction doesn't match the one of `SpanROI`, which follows the one of matplotlib `SpanSelector`.

Reply:
I see, however I don't see the issue as the code works as it should.

Reply:
Yes, very true and this is why this is just a suggestion to make it more consistent! This is not really important, because the widgets are not directly exposed in the public API, however someone  instantiating directly this kind of widgets may be surprised about this -at least I was!

Reply:
That's true. Maybe we could consider changing the definition of ``direction`` in the range widget to be consistent with that of Line in another PR?

Reply:
I would prefer the other way around, because the RangeWidget wraps a matplotlib SpanSelector, so if we need to choose between one the two definitions, I would prefer to stay consistent with the matplotlib one.

Reply:
But that would result in a horizontal ``VerticalLineWidget`` isn't it?

Reply:
Yes, indeed, a _horizontally moving_ `VerticalLineWidget`! :D